pkg/controller/attach_detach.go

```diff
@@ -218,11 +218,8 @@ func (r *ReconcileAttachDetach) triggerAttach(ctx context.Context, azVolumeAttac
 				return uerr
 			}
 		}
-		if derr := azureutils.UpdateCRIWithRetry(updateCtx, nil, r.client, r.azVolumeClient, azVolumeAttachment, updateFunc, consts.ForcedUpdateMaxNetRetry); derr != nil {
-			klog.Errorf("failed to update AzVolumeAttachment (%s) with attachVolume result (response: %v, error: %v): %v", azVolumeAttachment.Name, response, attachErr, derr)
-		} else {
-			klog.Infof("Successfully updated AzVolumeAttachment (%s) with attachVolume result (response: %v, error: %v)", azVolumeAttachment.Name, response, attachErr)
```

Comment:
Can we keep the log line for the successful updates? I think this is a good marker for tracing CRIs when debugging.

Reply:
Isn't the "Initiating..." trace from `azureutils.UpdateCRIWithRetry()` sufficient? I find these additional traces noisy and in some cases misleading. For example,

```log
2021-12-13T13:35:45.5541217Z E1213 13:31:30.162808       1 azvolume.go:166] failed to create volume CreateSnapshot-volume-3: rpc error: code = NotFound desc = Retriable: false, RetryAfter: 0s, HTTPStatusCode: 404, RawError: {"error":{"code":"ResourceNotFound","message":"The Resource 'Microsoft.Compute/disks/CreateSnapshot-volume-3' under resource group 'azuredisk-csi-driver-test-7332dc60-5c18-11ec-a1db-000d3ae6c0a6' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix"}}
2021-12-13T13:35:45.5545275Z I1213 13:31:30.162947       1 util.go:51] azure_disk_utils.go:580] Initiating update with retry for *v1alpha1.AzVolume (createsnapshot-volume-3)
2021-12-13T13:35:45.5547191Z I1213 13:31:30.169979       1 azvolume.go:191] successfully created volume (CreateSnapshot-volume-3) with volume creation result (response: <nil>, error: <nil>)
```

The last trace makes it seem like the operation succeeded when it actually failed. (I could fix this trace, I suppose, but I don't find these traces particularly useful.)

Reply:
I usually referred to these logs to confirm whether CRI updates succeeded or not when CRIs were stuck in transient states but yeah, I do see that the log message is misleading and that we would need to edit the log lines if we were going to continue using it.

Reply:
An error will be logged by `azureutils.UpdateCRIWithRetry` if the update fails, otherwise we can assume success.

Reply:
Yes, I guess I felt the need for a searchable log line indicating success given that many operations are done concurrently making the log file not so easy to follow but I guess we can search for update failure errors instead and assume success if none was found.